assert/assertions_test.go

```diff
@@ -1338,31 +1338,70 @@ func TestPanicsWithValue(t *testing.T) {
 }
 
 func TestPanicsWithError(t *testing.T) {
+	type NestedErr struct {
+		Err error
+	}
 
-	mockT := new(testing.T)
-
+	mockT := new(CollectT)
```

Comment:
Do not use CollectT for that purpose. `CollectT` is reserved for use with `EventuallyWithT`.

`captureTestingT` is probably what you need instead.